src/main/java/org/tron/p2p/utils/NetUtil.java

```diff
@@ -195,4 +204,30 @@ public static InetSocketAddress parseInetSocketAddress(String para) {
           + "use ipv4:port or [ipv6]:port", para));
     }
   }
+
+  private static String getIp(List<String> multiSrcUrls) {
+    ExecutorService executor = Executors.newCachedThreadPool();
+    CompletionService<String> completionService = new ExecutorCompletionService<>(executor);
+
+    List<Callable<String>> tasks = new ArrayList<>();
+    multiSrcUrls.forEach(url -> tasks.add(() -> getExternalIp(url)));
+
+    for (Callable<String> task : tasks) {
+      completionService.submit(task);
+    }
+
+    Future<String> future;
+    String result = null;
+    try {
+      future = completionService.take();
+      result = future.get();
+    } catch (InterruptedException | ExecutionException e) {
+      //ignore
+    } finally {
+      executor.shutdownNow();
+    }
+
+    return result;
```

Comment:
If there is an exception in obtaining the IP, should we consider returning a default value?

Reply:
The logic is same as before. If it returns null, the result should be checked by upper business.